src/Psl/DateTime/DateTime.php

```diff
@@ -159,96 +142,113 @@ public static function todayAt(int $hours, int $minutes, int $seconds = 0, int $
      */
     public static function fromParts(int $year, Month|int $month, int $day, int $hours = 0, int $minutes = 0, int $seconds = 0, int $nanoseconds = 0, Timezone $timezone = null): self
```

Comment:
This is the only static constructor where timezone is optional.
How do you see it : always explicitly provide it (even though a lot of the times you just want the default one) or optionally provide it?

Reply:
actually i missed this, it should be required.

I don't know how we can define a default timezone without touching the global state.

Reply:
timezone is now optional in all static functions, we use Timezone::default(), but this means these functions can't be pure.